dotnet/src/SemanticKernel.Abstractions/Orchestration/ContextVariables.cs

```diff
@@ -26,10 +26,26 @@ public sealed class ContextVariables : IDictionary<string, string>
     /// <summary>
     /// Constructor for context variables.
     /// </summary>
-    /// <param name="value">Optional value for the main variable of the context including trust information.</param>
-    public ContextVariables(string? value = null)
+    /// <param name="input">Optional value for the main variable of the context including trust information.</param>
+    /// <param name="args">Optional list of key-value pairs to initialize the context with.</param>
+    public ContextVariables(
+        string? input = null,
+        IEnumerable<KeyValuePair<string, string>>? args = null)
     {
-        this._variables[MainKey] = value ?? string.Empty;
+        this.Set(MainKey, input ?? string.Empty);
```

Comment:
Why change a null value to an empty string?

Reply:
It looks like if the value is null, the variable will be deleted. But I believe there should be a better way how to handle null-empty scenarios. Probably out of scope of this PR.

Reply:
Yeah, I did simplify it just to 'null', and it ended up breaking a bunch of UTs. I figured it'd be best to fix that case and all of its UTs in another change. Also, we should expect the "input" to move out of this class soon. In principle, I agree that null strings should remain null, if not provided.

Reply:
Created Issue to track:
https://github.com/microsoft/semantic-kernel/issues/2216